frontend/src/components/grid_CLASS/grid.jsx

```diff
@@ -2,13 +2,13 @@ import React from 'react';
 import GridRow from './grid_row';
 
 class Grid extends React.Component {
-  componentDidMount() {
-    // if (!this.props.game) this.props.fetchNewGame();
-    document.addEventListener('keydown', this.handleGuess.bind(this));
-  }
+  // componentDidMount() {
+  //   // if (!this.props.game) this.props.fetchNewGame();
+  //   document.addEventListener('keydown', this.handleGuess.bind(this));
+  // }
 
-  componentWillMount() {
-    document.removeEventListener('keydown', this.handleGuess.bind(this));
+  componentWillUnmount() {
+    // document.removeEventListener('keydown', this.handleGuess.bind(this));
     this.props.clearGameState();
```

Comment:
Won't this clear out the game state on _every_ re-render?

Reply:
nope, componentWillUnmount only runs when the game page is exited